notebooks/video_clips/VideoClips.ipynb

```diff
@@ -0,0 +1,963 @@
+    "    namesfile=\"models/coco.names\" # path for output to name mapping\n",
+    "    input_size=416\n",
+    "    use_gpu=False # use GPU or not\n",
+    "    outdir=\"output/norman\"\n",
+    "    no_squash_detections=True # if detections exist, don't rerun.\n",
+    "    def __init__(self, image='',stream=''):\n",
+    "        self.image = image\n",
+    "        self.stream=stream # 'webcam' to open webcam w/ OpenCV\n",
+    "\n",
+    "# now we pull data, and and run detection.\n",
+    "dopts = DetectorOptions( stream=\"norman.mp4\")\n",
+    "yolo = RemoteYOLOv4.__new__(RemoteYOLOv4)\n",
+    "yolo.__init__(dopts)\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "fa362d75-8c06-4adb-a0ba-3b66421a1807",
+   "metadata": {},
+   "source": [
+    "## Now Check Detections\n",
+    "the YoloV4 code we use outputs detections sequentially into a csv file, so let's load the file and see what the output looks like."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "e3053bdb-a78a-4a8e-a454-2507a1477eb9",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import pandas as pd\n",
+    "df = pd.read_csv(\"output/norman/detections.csv\")\n",
+    "print(df)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "5a8cd9e9-ecaf-4e67-896b-019a6c30f1c3",
+   "metadata": {},
+   "source": [
+    "# Process Into Clips\n",
+    "Now that we've verified that we have the data from the model, we will take the output and process it into Clips.\n",
+    "\n",
+    "We will define a few classes and functions to process our model output.\n",
+    "\n",
+    "- `ClipOptions` - an options class that we will use to define how it works;  \n",
+    "- `preprocess` - convert the dataframe into information the detection can use\n",
+    "- `Clip` - a class that defines our output\n",
+    "- `ClipStorage` - a class for maintaining state between the functions\n",
+    "- `process_new_frame` - a function to run when we see a new frame\n",
+    "- `process_row` - a function to run we we see a new detection\n",
+    "- `process` - the function to process the whole video/csv.\n",
+    "\n",
+    "These could be in a single class, but I've left them apart to allow people to take them piece by piece."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "2503ff66-92c7-46a6-9dfb-9b55882c79bf",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import logging\n",
+    "# Fist we'll define the options we're going to use.\n",
+    "class ClipOptions:\n",
+    "    offset_frame=0 # starting offset in frames\n",
+    "    end_frame=-1 # ending offset in frames\n",
+    "    initconf=50 # minimun confidence to start ( 0-100 )\n",
+    "    initlen=5 # minimum detection duration in frames to start a clip\n",
+    "    dropconf=25 # confidence to end a frame (0 -100 )\n",
+    "    droplen=5 # number of detection missed frames to end a clip\n",
```

Comment:
There is a lot of helper code that's not showing anything Aperture related and distracting. It can be imported in the dependencies sections via a helper file

Reply:
ALso take a look at this from our previous notebook that @gsaluja9  built to wrap Clips into DataModel classes in Python SDK. I believe you will need to enhance them to remove all this JSON creation from this notebook and we will get more useful code in the Python SDK